### PR TITLE
DAOS-623 build: Avoid calling pkg-config unnecessarily

### PR DESCRIPTION
**Pull request by jolivier23**
When there is no 'lib' or 'lib64' directory under component prefix,
the call to ParseConfig is unlikely to succeed so avoid trying it.
This removes error messages from pkg-config from the build output
and avoids unecessary invocations.

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

Required-githooks: true

### CODE REVIEW

File: site_scons/prereq_tools/base.py [hunk 1]
```diff
@@ -1470,24 +1470,26 @@ def has_missing_system_deps(self, env):
...
         if self.pkgconfig is None:
-            return False
-
+            return
         path = os.environ.get("PKG_CONFIG_PATH", None)
         if path and "PKG_CONFIG_PATH" not in env["ENV"]:
             env["ENV"]["PKG_CONFIG_PATH"] = path
-        if self.component_prefix:
+        if not self.use_installed and not self.component_prefix == "/usr":
+            path_found = False
```
Comment (ashleypittman) [inline site_scons/prereq_tools/base.py:1478]: Do you mean to set this to true at any point?

File: site_scons/prereq_tools/base.py [hunk 2]
```diff
@@ -1470,24 +1470,26 @@ def has_missing_system_deps(self, env):
...
             for path in ["lib", "lib64"]:
                 config = os.path.join(self.component_prefix, path, "pkgconfig")
                 if not os.path.exists(config):
                     continue
                 env.AppendENVPath("PKG_CONFIG_PATH", config)
+            if not path_found:
+                return
```
Comment (daltonbohning) [inline site_scons/prereq_tools/base.py:1485]: Continued from Ashley's comment, this would throw an exception if `path_found` is not defined

**Comment by jolivier23** [comment]
I realize now that for daos, pkgconfig isn't actually necessary here but PKG_CONFIG_PATH needs to be set so mercury will use ucx

File: site_scons/prereq_tools/base.py (hunk 2 above)
Comment (daltonbohning) [reply site_scons/prereq_tools/base.py:1485]: Ah, this is under the same `if` scope. Please ignore this comment. But I think Ashley's comment above is still valid.

**Review by daltonbohning — APPROVED** [review]

**Review by ashleypittman — APPROVED** [review]

**Pull request merged**